lib/more_core_extensions/core_ext/enumerable/deep_clone.rb

```diff
@@ -0,0 +1,21 @@
+module MoreCoreExtensions
+  module EnumerableDeepClone
+    # Create a deep clone of the enumerable object. This is similar to deep_dup
+    # but uses a Marshal based approach instead.
+    #
+    #   h1 = {:a => "hello"}
+    #   h2 = h1.deep_clone
+    #
+    #   h1[:a] << " world"
+    #
+    #   h1[:a] # "hello world"
+    #   h2[:a] # "hello"
+    #
+    def deep_clone
```

Comment:
What do you think of `#deep_clone` living in https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/shared/nested.rb instead?

Reply:
Would putting it there mean that users would have to explicitly include it to use it? I guess I'm not entirely clear on that particular module's purpose.

Reply:
It is meant for nested objects like [hashes](https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/hash/nested.rb#L5) and [arrays](https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/array/nested.rb#L5).  It feels a little more specific than all of enumerable.

Reply:
@bdunne ok, updated.